src/statsbase.jl

```diff
@@ -0,0 +1,186 @@
+for fun in [:var, :std]
+    @eval function Statistics.$fun(A::KeyedArray, wv::AbstractWeights; dims=:, corrected=true, kwargs...)
+        dims === Colon() && return $fun(parent(A), wv; kwargs...)
+        numerical_dims = AxisKeys.hasnames(A) ? NamedDims.dim(dimnames(A), dims) : dims
+        data = $fun(parent(A), wv, numerical_dims; corrected=corrected, kwargs...)
+        new_keys = ntuple(d -> d in numerical_dims ? Base.OneTo(1) : axiskeys(A,d), ndims(A))
+        return KeyedArray(data, map(copy, new_keys))#, copy(A.meta))
+    end
+
+    @eval function Statistics.$fun(A::KeyedArray, wv::KeyedVector; kwargs...)
+        return $fun(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+    end
+end
+
+function Statistics.cov(A::KeyedMatrix, wv::AbstractWeights; dims=1, corrected=true, kwargs...)
+    # A little awkward, but the weighted `cov` method from statsbase only works
+    # on dense matrices, so we need to unwrap this twice and completely rebuild the
+    # array
+    if AxisKeys.hasnames(A)
+        numerical_dim = NamedDims.dim(dimnames(A), dims)
+        data = cov(parent(parent(A)), wv, numerical_dim; corrected=corrected, kwargs...)
+        rem_dim = first(setdiff((1, 2), numerical_dim))
+        rem_name = dimnames(A, rem_dim)
+        rem_key = axiskeys(A, rem_dim)
+        return KeyedArray(
+            NamedDimsArray(data, (rem_name, rem_name)),
+            (copy(rem_key), copy(rem_key))
+        )
+    else
+        data = cov(parent(A), wv, dims; corrected=corrected, kwargs...)
+        # Use same remaining axis for both dimensions of data
+        rem_dim = first(setdiff(ndims(A), dims))
+        new_keys = Tuple(copy(axiskeys(A, rem_dim)) for i in 1:2)
+        return KeyedArray(data, new_keys)
+    end
+end
+
+function Statistics.cov(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return cov(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+function Statistics.cor(A::KeyedMatrix, wv::AbstractWeights; dims=1, kwargs...)
+    # A little awkward, but the weighted `cov` method from statsbase only works
+    # on dense matrices, so we need to unwrap this twice and completely rebuild the
+    # array
+    if AxisKeys.hasnames(A)
+        numerical_dim = NamedDims.dim(dimnames(A), dims)
+        data = cor(parent(parent(A)), wv, numerical_dim; kwargs...)
+        rem_dim = first(setdiff((1, 2), numerical_dim))
+        rem_name = dimnames(A, rem_dim)
+        rem_key = axiskeys(A, rem_dim)
+        return KeyedArray(
+            NamedDimsArray(data, (rem_name, rem_name)),
+            (copy(rem_key), copy(rem_key))
+        )
+    else
+        data = cor(parent(A), wv, dims; kwargs...)
+        # Use same remaining axis for both dimensions of data
+        rem_dim = first(setdiff((1, 2), dims))
+        new_keys = Tuple(copy(axiskeys(A, rem_dim)) for i in 1:2)
+        return KeyedArray(data, new_keys)
+    end
```

Comment:
There has got to be a more compact pattern for these things. Maybe it's like this, will this work? 
```suggestion
    d = NamedDims.dim(dimnames(A), dims)
    data = scattermat(unname(keyless(A)), wv...; dims=d, kwargs...)
    L1 = dimnames(A, 3-d)
    data2 = hasnames(A) ? NamedDimsArray(data, (L1, L1)) : data
    K1 = axiskeys(A, 3-d)
    KeyedArray(data2, (copy(K1), copy(K1)))
```
Perhaps there should be a unified function for doing `unname(keyless`, but probably still pretty cheap.

Reply:
I made one here: https://github.com/mcabbott/AxisKeys.jl/commit/82d7037de5e99c5e2d48594de76dd3b0780c83c2